aws-s3-transfer-manager/src/operation/download.rs

```diff
@@ -172,11 +172,11 @@ async fn send_discovery(
         discovery.chunk_meta,
     );
 
-    if !discovery.remaining.is_empty() {
+    if let Some(remaining) = discovery.remaining {
```

Comment:
What if the Range is Some(empty)? Will the distribute_work function work correctly?

Reply:
The docs says that "When using an inclusive range for iteration, the values of start() and [end()](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html#method.end) are unspecified after the iteration ended. To determine whether the inclusive range is empty, use the [is_empty()](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html#method.is_empty) method instead of comparing start() > end()." 

so we should keep the is_empty check as well or move it to the distribute_work function.